sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -1151,10 +1151,21 @@ def extend(
         # When a corrupt value made it into the cache, we have to fail.
         raise Exception("Unexpected cached value: %s" % cached_value)
     # Write to state handler
+    futures = []
     out = coder_impl.create_OutputStream()
     for element in elements:
       coder.encode_to_stream(element, out, True)
-    return self._underlying.append_raw(state_key, out.get())
+      if out.size() > data_plane._DEFAULT_SIZE_FLUSH_THRESHOLD:
+        futures.append(self._underlying.append_raw(state_key, out.get()))
+        out = coder_impl.create_OutputStream()
+    if out.size():
+      futures.append(self._underlying.append_raw(state_key, out.get()))
+    return _DeferredCall(
```

Comment:
if there is a single future, is it worth returning it directly instead of the gather?

Reply:
I thought about that, but I think the savings is negligible and this means that the base case exercises the same code path as the exceptional one.